__tests__/SetsSelectors.test.js

```diff
@@ -820,6 +820,286 @@ describe('SetsSelectors', () => {
+                                isValid: true,
+                                removed: false,
+                                data: [-3456, 37, 1.833368, 388]
+                            }],
+                        },
+                        e: {
+                            exercise: 'Bench',
+                            weight: 100,
+                            metric: 'lbs',
+                            reps: [{
+                                isValid: true,
+                                removed: false,
+                                data: [-3456, 37, 1.3, 388]
+                            }, 
+                            {
+                                isValid: true,
+                                removed: false,
+                                data: [-3456, 37, 1.9, 388]
+                            },
+                            {
+                                isValid: true,
+                                removed: false,
+                                data: [-3456, 37, 2.4, 388]
+                            }
+                        ]},
+                    }
+                }
+            };
+
+            let set = {
+                exercise: 'Bench',
+                weight: 100,
+                metric: 'lbs',
+                reps: [{
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 37, 1.333368, 388]
+                }, 
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 37, 1.033368, 388]
+                },
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 37, 1.453368, 388]
+                }],
+            };
+
+            let result = sut.getBestAvgVelocityEver(state, set);
+
+            expect(result).toBe(2.4);
+        });
+
+        test('return null when no history data', () => {
+            const state = {
+                sets: {
+                    historyData: {}
+                }
+            };
+
+            let set = {
+                exercise: 'Bench',
+                weight: 100,
+                metric: 'lbs',
+                reps: [{
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 37, 1.333368, 388]
+                }, 
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 37, 1.033368, 388]
+                },
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 37, 1.453368, 388]
+                }],
+            };
+
+            let result = sut.getBestAvgVelocityEver(state, set);
+
+            expect(result).toBe(null);
+        });
+    });
+
+    describe('getBestROMEver', () => {
+        
+                test('get best rom ever', () => {
```

Comment:
- [ ]  modify test to have a larger value as invalid and another larger value as removed, should still work